src/main/java/com/esri/arcgisruntime/toolkit/skins/FeatureTemplatePickerTilePaneSkin.java

```diff
@@ -0,0 +1,329 @@
+        Toggle matchingToggle = toggleGroup.getToggles().stream()
+            .filter(t -> (t.getUserData()).equals(newValue))
+            .findFirst()
+            .orElse(null);
+        toggleGroup.selectToggle(matchingToggle);
+      }
+    });
+
+    // add a tile pane for each feature template group to the view
+    control.getFeatureTemplateGroups().stream()
+        .map(this::createVBoxForFeatureTemplateGroup)
+        .forEach(tilePane -> contentPane.getChildren().add(tilePane));
+
+    // add or remove tile panes if the feature template group collection changes
+    control.featureTemplateGroupsProperty().addListener((ListChangeListener<FeatureTemplateGroup>) c -> {
+      while (c.next()) {
+        if (c.wasAdded()) {
+          // create vboxes for the each feature template group
+          List<VBox> added = c.getAddedSubList().stream()
+              .map(this::createVBoxForFeatureTemplateGroup)
+              .collect(Collectors.toList());
+          // add the vboxes to the view
+          contentPane.getChildren().addAll(c.getFrom(), added);
+        } else if (c.wasRemoved()) {
+          // remove the vboxes based on the
+          contentPane.getChildren().remove(c.getFrom(), c.getFrom() + c.getRemovedSize());
+        }
+      }
+    });
+
+    control.orientationProperty().addListener((observable, oldValue, newValue) ->
+      this.updateOrientation()
+    );
+
+    // update all swatches together when the symbol size changes
+    control.symbolSizeProperty().addListener((observable, oldValue, newValue) -> {
+      swatchesNeedUpdate = true;
+      control.requestLayout();
+    });
+
+    // update all item swatches once populated
+    swatchesNeedUpdate = true;
+    control.requestLayout();
+  }
+
+  @Override
+  protected void layoutChildren(double contentX, double contentY, double contentWidth, double contentHeight) {
+    if (swatchesNeedUpdate) {
+      // ensure an update is only triggered once per symbol size change
+      swatchesNeedUpdate = false;
+      // trigger all symbol swatch updates
+      CompletableFuture<?>[] futures = contentPane.getChildren().stream()
+          .flatMap(n -> ((VBox) n).getChildren().stream())
+          .filter(n -> n instanceof TilePane)
+          .flatMap(tilePane -> ((TilePane) tilePane).getChildren().stream())
+          .filter(n -> n instanceof ToggleButton)
+          .map(n -> {
+            ToggleButton toggleButton = (ToggleButton) n;
+            FeatureTemplateItem featureTemplateItem = (FeatureTemplateItem) toggleButton.getUserData();
+            ImageView imageView = (ImageView) toggleButton.getGraphic();
+            return updateSwatch(featureTemplateItem, imageView);
+          })
+          .toArray(CompletableFuture<?>[]::new);
+      // wait until all updates finished before updating layout normally
+      CompletableFuture.allOf(futures).thenRunAsync(() ->
+        super.layoutChildren(contentX, contentY, contentWidth, contentHeight)
```

Comment:
What thread does this run on? I assume it should be running on the UI thread.

Reply:
Good call. I decided to call `requestLayout` here instead to ensure it's run on the JavaFX thread.